### DISCUSSION TITLE
How to correctly highlight it?

### DISCUSSION THREAD

**Discussion by ghost** [orbitalquark/textadept#369] [General]
The keyword `Nil` on the language has two meanings. It's both a constant and a type (the language doesn't use `Void` and reused `Nil` for Void type). TextAdept doesn't have the power to parse the syntax of the language to correctly determine when `Nil` is a constant and when `Nil` is a type. This is really difficult to do.

**Comment by orbitalquark** [comment]
You could try a contextual match. I don't know your exact syntax, but if 
a `Nil` type is followed by whitespace and then a word, you could match 
that and highlight it was a type. Otherwise, highlight it was a constant.
